Source/ASCollectionView.mm

```diff
@@ -916,7 +916,9 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
   ASCellNode *cell = [self nodeForItemAtIndexPath:indexPath];
   if (cell.shouldUseUIKitCell) {
     if ([_asyncDelegate respondsToSelector:@selector(collectionView:layout:sizeForItemAtIndexPath:)]) {
-      return [(id)_asyncDelegate collectionView:collectionView layout:collectionViewLayout sizeForItemAtIndexPath:indexPath];
+      CGSize size = [(id)_asyncDelegate collectionView:collectionView layout:collectionViewLayout sizeForItemAtIndexPath:indexPath];
+      cell.style.preferredSize = size;
```

Comment:
@maicki could you share ideas on alternatives? Your feedback matters a lot!!

The idea here is that the app is not providing an ASCellNode at all, so it's not possible for the app to set the preferredSize. It never has access to the object.

The best possible implementation would be adjusting ASDataController to be able to use something like NSPointerArray, or [NSNull null], to indicate UICollectionView passthrough cells.  Since this isn't supported today, this ASCellNode is used as a very simple (low code complexity) mechanism to store & cache the synchronously calculated layout information.

This feature has ended up being very useful for apps looking for a migration path to AsyncDisplayKit, which is awesome considering how little code is needed to support the passthrough of a UICollectionViewCell. But we should think about whether it's worth the effort for a larger refactor to support this.